Ensure Grails plugin `assemble` task produces only a JAR artifact

matrei opened the pull request:
Previously, running the `assemble` task in a Grails plugin project only generated `*.war` and `*-plain.war` artifacts. This change ensures that the task also produces the expected JAR artifact, aligning with typical plugin packaging needs.

matrei committed:
fix: ensure Grails plugin `assemble` task produces a JAR artifact

Previously, running the `assemble` task in a Grails plugin project only generated `*.war` and `*-plain.war` artifacts. This change ensures that the task also produces the expected JAR artifact, aligning with typical plugin packaging needs.

matrei committed:
refactor: name task configuration closure parameters for clarity

codeconsole reviewed (changes requested):
Will the tasks `bootWar, bootDistZip, bootDistTar, bootStartScripts, distZip, distTar` still continue to run? if so, should they be disabled as well?

why generate a war and a war-plain? should that be disabled as well?

Probably best in tbe end, bootRun should work and assemble should just produce 1 jar.

matrei committed:
fix: configure `assemble` to produce only a plain JAR

Disable tasks for bootable artifacts (e.g., `bootJar`, `bootWar`, etc.) and other outputs like war.
